swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -372,7 +372,10 @@ private void mergeWithExistingSubscriptionRecords(
                 existingSubscription);
             subscriptionRepository.delete(existingSubscription);
           } else {
-            if (!matchingUpdatedSubscription.equals(existingSubscription)) {
+            if (!matchingUpdatedSubscription.equals(existingSubscription)
+                || !Objects.equals(
```

Comment:
Subscription excludes subscriptionMeasurements from the .equals so adding it here.